src/classes/HH_Households_TDTM.cls

```diff
@@ -239,7 +239,7 @@ public without sharing class HH_Households_TDTM extends TDTM_Runnable {
             
             if (householdnameupdates.size() > 0){
                               
-                if (currentHouseholdsSettings.npo02__Async_Household_Naming__c == true)
+                if (currentHouseholdsSettings.npo02__Async_Household_Naming__c == true && !system.isFuture() && !system.isBatch())
```

Comment:
it is an existing bug I ran into while developing Manage Household UI.  since it has been around forever (since 2.x), I felt it wasn't critical to get in sooner, and could just go in with my work and save the time of creating yet another branch, etc.